include/rocksdb/utilities/backupable_db.h

```diff
@@ -262,18 +270,20 @@ class BackupEngine {
   // avoid losing unflushed key/value pairs from the memtable.
   virtual Status CreateNewBackupWithMetadata(
       DB* db, const std::string& app_metadata, bool flush_before_backup = false,
-      std::function<void()> progress_callback = []() {}) = 0;
+      std::function<void()> progress_callback = []() {},
+      const CreateBackupOptions& opt = CreateBackupOptions()) = 0;
```

Comment:
My preference is to create a new function
`virtual Status CreateNewBackupWithMetadata(const CreateBackupOptions& opt, DB* db, const std::string& app_metadata)` and put all other arguments into `CreateBackupOptions` and keep the current one by doing a simple translation.